### PR TITLE
Add test for hibernate-reactive

### PR DESCRIPTION
**Pull request by laurit**
Add test for hibernate-reactive that builds on vertx-sql-client

### CODE REVIEW

File: instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxImplInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+package io.opentelemetry.javaagent.instrumentation.vertx.v4_0.sql;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+public class VertxImplInstrumentation implements TypeInstrumentation {
```
Comment (mateuszrzeszutek) [inline instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxImplInstrumentation.java:15]: This does not seem to be used

File: instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/ConnectionFactoryBaseInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.vertx.v4_0.sql;
+
+public class ConnectionFactoryBaseInstrumentation {
```
Comment (mateuszrzeszutek) [inline instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/ConnectionFactoryBaseInstrumentation.java:8]: This is also unused

File: instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxImplInstrumentation.java (hunk 1 above)
Comment (laurit) [reply instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxImplInstrumentation.java]: sorry, forgot to remove failed approaches

File: instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxSqlClientSingletons.java [hunk 3]
```diff
@@ -74,5 +78,19 @@ public static Scope endQuerySpan(Map<Object, Object> contextData, Throwable thro
...
 
+  private static final VirtualField<SqlClientBase<?>, SqlConnectOptions> connectOptionsField =
+      VirtualField.find(SqlClientBase.class, SqlConnectOptions.class);
+
+  public static Future<SqlConnection> attachConnectOptions(
+      Future<SqlConnection> future, SqlConnectOptions connectOptions) {
+    return future.map(
+        sqlConnection -> {
+          if (sqlConnection instanceof SqlClientBase) {
+            connectOptionsField.set((SqlClientBase<?>) sqlConnection, connectOptions);
```
[the remark this answers is not in the document]
Comment (laurit) [reply instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxSqlClientSingletons.java:90]: same field is used in `SqlClientBaseInstrumentation` I added a comment for that

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxSqlClientSingletons.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mateuszrzeszutek) [reply instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxSqlClientSingletons.java:90]: Thanks!

**Comment by trask** [comment]
should we add to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks?

**Comment by laurit** [comment]
> should we add to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks?

Currently there is no hibernate-reactive specific instrumentation, these are just tests that verify that at least for a simple query it works as expected. This far we have not listed frameworks for which we don't have dedicated instrumentation.

**Comment by trask** [comment]
oh, I see, I saw that new instrumentation was added, but that's just making vertx-sql-client instrumentation work better 👍

**Pull request merged**